waspc/data/Generator/templates/sdk/wasp/auth/useAuth.ts

```diff
@@ -21,7 +22,8 @@ function createUserGetter() {
     try {
       const response = await api.get(getMeRoute.path)
   
-      return superjsonDeserialize(response.data)
+      const userData = superjsonDeserialize<AuthUserData | null>(response.data)
+      return userData ? enrichAuthUser(userData) : null;
```

Comment:
Hm, I feel like there's a better name for this than `enrich`, especially considering this function maps one data type to another (i.e., it doesn't only enrich a value through side effects, it performs a transformation).

Even if there isn't a standard name for this operation (other than `serialize` which would obviously be confusing :)), I'd still prefer something simpler like `makeAuthUser` or `getAuthUser` of type `(data: AuthUserData) => AuthUser`.

Perhaps even better, an `AuthUser` could be a class that takes `AuthUserData` through its constructor. This way, we'd have `new AuthUser(userData)`.

Reply:
Enrich felt okay since it turned one _plain thing_ into this new _cool thing_ 😄 

The suggestion to use a class sounds good to me, I'll try doing that and see where it leads me.

Reply:
Using a class seemed complex to maintain e.g. all the props in the object would need to be maintained as class properties. I went with a simpler name i.e. `makeAuthUser` :)